staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -238,11 +249,26 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 			}
 		}
 	}
+
 	// if there's no name on the request and we thought it was a delete before, then the actual verb is deletecollection
 	if len(requestInfo.Name) == 0 && requestInfo.Verb == "delete" {
 		requestInfo.Verb = "deletecollection"
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AuthorizeWithSelectors) {
+		if !verbViaPathPrefix && (requestInfo.Verb == "list" || requestInfo.Verb == "watch" || requestInfo.Verb == "deletecollection") {
```

Comment:
`verbViaPathPrefix` is an interesting exclusion.  I guess I can't think of a special verb case where I'd want the selectors

Reply:
watch and proxy were the only two special verbs, and the single-item watch endpoint that doesn't support selectors really threw things for a loop... I excluded it because I didn't feel bad telling anyone who wants the benefit of narrower authz to use the non-deprecated watch endpoint